### ISSUE TITLE
Reset to default settings deletes all your aliases

### ISSUE THREAD

**Issue by ovalseven8** [qTox/qTox#3664]
##### Brief Description

OS: Ubuntu 16.04
Commit hash: https://github.com/qTox/qTox/commit/fcfac15c785f6cca5a38b9aa4f915fb805aec161 

Reproducible: Always
##### Steps to reproduce
1. Go to settings -> Advanced -> Click "Reset to default settings"
2. Restart qTox
3. All your set aliases in the contact list are deleted!

**Comment by Diadlo** [comment]
This button remove all user changes, include aliases

**Comment by ovalseven8** [comment]
Oh, that's a bad idea.

**Comment by sudden6** [comment]
I think, the behavior mentioned by @Diadlo is good, many applications provide this sort of cleanup button.

**Comment by zetok** [comment]
@sudden6 nope. Removing **user data** is never a good thing when one wants to just reset **settings**..

**Comment by Diadlo** [comment]
**NOTE:** Who will fix this bug also should remove information about it from [doc/user_manual_en.md](https://github.com/qTox/qTox/blob/master/doc/user_manual_en.md#advanced)

**Comment by AliceGrey** [comment]
From the looks of it, it calls `Settings::resetToDefault()` which basically deletes `username.ini` which contains the settings for the user but leaves the `username.tox` file intact. I haven't gone through a whole lot of the settings code to see how it works but it looks like at the end of `Settings::resetToDefault()` you could call `Settings::getInstance().createPersonal(username);` which should generate a new `username.ini` file with the default settings, allowing the user to be used again after hitting the reset button. Is there anyone who is more familiar with the Settings code that can confirm that this would work properly?

**Comment by zetok** [comment]
@AliceGrey that would be kinda a work-around – i.e. alias data should go into DB, not the .ini files.

**Comment by sudden6** [comment]
@AliceGrey There is a problem with the approach of keeping the profile data only in RAM. When the `username.ini` file is deleted, the data exists only in the RAM. If there's a power outage/crash/whatever at this very moment the data will be lost.This can be avoided by, for example, creating a new file with the appropriate content first, then renaming it to the old files name.

Since renaming is an atomic operation at OS level (normally) there's no point where the data isn't stored on the disk. Note that this doesn't take the OS file system cache into account which also prevents data from being stored on disk, but I think you get the idea.

**Comment by AliceGrey** [comment]
@sudden6 So basically the `Settings::resetToDefault()` function needs to be reworked so it creates the new .ini file from the default settings before it deletes the current one.

 @zetok It looks like currently the reset button only calls `Settings::resetToDefault()` which currently just finds and deletes the `username.ini` file and doesn't touch anything in the DB or `username.tox` file. Does the DB hold user settings in it?

**Comment by Diadlo** [comment]
@AliceGrey We don't need create new settings file on reset. It will be created automatically on next startup

**Comment by zetok** [comment]
@AliceGrey well, aliases aren't user settings to begin with, but user data. And user data should be stored in the DB. Even though part of user data is right now stored in settings :s

And fix to this issue should do exactly that – move user data from settings to the DB.
